### ISSUE TITLE
Transaction has been declined. Please try again later.

### ISSUE THREAD

**Issue by manjula-gurjar-bod** [AbsoluteWebServices/vsf-amazon-pay#3]
I am getting a transaction declined error while placing the order and also it displays the amazon pay method more than one time on the VSF checkout page. FYI I am using Magento 2.4.1 so is there any version difference in the API?

**Comment by dmitry-kozachek** [comment]
Hi @manjula-gurjar-bod, please have a look on this [answer](https://github.com/AbsoluteWebServices/vsf-amazon-pay/issues/2#issuecomment-610218828), we doesn't provide magento extension, so if you don't have one - you need to get it somewhere

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek Magento already has an amazon pay payment method module by default but this module shows multiple amazon pay options on checkout and also not proceeding the order or directly place an order without open the amazon payment gateway page.

**Comment by dmitry-kozachek** [comment]
@manjula-gurjar-bod I don't think this vsf module will work with default amazon pay payment method. It's not an addition to default amazon pay payment method.
Also, like I said in this [answer](https://github.com/AbsoluteWebServices/vsf-amazon-pay/issues/2#issuecomment-610218828) - this module isn't tested.
Also this module requires you to make some theme changes and use module [components](https://github.com/AbsoluteWebServices/vsf-amazon-pay/tree/master/components). You need to use [Button](https://github.com/AbsoluteWebServices/vsf-amazon-pay/blob/master/components/Button.vue) component to initialize amazon pay, then you need to use [AddressBook](https://github.com/AbsoluteWebServices/vsf-amazon-pay/blob/master/components/AddressBook.vue) and [Wallet](https://github.com/AbsoluteWebServices/vsf-amazon-pay/blob/master/components/Wallet.vue) on appropriate checkout steps.

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek 
        "GetOrderReferenceDetails": "http://localhost:8080/api/ext/amazon-pay/GetOrderReferenceDetails",
        "SetOrderReferenceDetails": "http://localhost:8080/api/ext/amazon-pay/SetOrderReferenceDetails",
        "ConfirmOrderReference": "http://localhost:8080/api/ext/amazon-pay/ConfirmOrderReference",
        "CancelOrderReference": "http://localhost:8080/api/ext/amazon-pay/CancelOrderReference",
        "Authorize": "http://localhost:8080/api/ext/amazon-pay/Authorize"

**What do we need to do for these endpoints? As I am getting 404 after selecting the address in the address book for http://localhost:8080/api/ext/amazon-pay/GetOrderReferenceDetails this URL.**

**Comment by dmitry-kozachek** [comment]
@manjula-gurjar-bod Seems like you have not installed vsf-api extension as it it described here https://github.com/AbsoluteWebServices/vsf-amazon-pay#amazon-pay-payment-api-extension

You can find this extension in the [API](https://github.com/AbsoluteWebServices/vsf-amazon-pay/tree/master/API) folder

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek I already have an amazon-pay folder in the vsf-api by the copy command described  cp -f ./vue-storefront/src/modules/vsf-amazon-pay/API/amazon-pay ./vue-storefront-api/src/api/extensions/

Do I need to make any other changes also or need to install anything else? Because getting 404 for http://localhost:8080/api/ext/amazon-pay/GetOrderReferenceDetails this URL.

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek  I am getting **The currency code 'GBP' passed into the request doesn't match the currency code in the OrderTotal ''** this error at the time of order placement and it stuck at payment polling page (https://payments.amazon.co.uk/checkout/widgets/paymentPolling?) after this order is visible in amazon pay seller central but not having merchant order id and also not saving in the Magento.

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek how can we pass the SCA after order placement?

**Comment by dmitry-kozachek** [comment]
@manjula-gurjar-bod please see answer to this question #2. This vsf module require you to handle backend part of payment processing

**Comment by manjula-gurjar-bod** [comment]
@dmitry-kozachek that backend part we need to do in magento?

**Comment by dmitry-kozachek** [comment]
@manjula-gurjar-bod yes
